File: csharp/sum-of-multiples/SumOfMultiples.cs
using System.Collections.Generic;
using System.Linq;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max) =>
        Enumerable.Range(1, max - 1)
            .Where(a => multiples.Any(b => a.IsMultipleOf(b)))
            .Sum();
}

internal static class Extensions
{
    public static bool IsMultipleOf(this int self, int other) => other switch
    {
        0 => false,
        _ => self % other == 0
    };
}

File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  def initialize(*numbers)
    @numbers = numbers
  end

  def to(limit)
    (1..limit - 1).select { |i| multiple?(i) }.sum
  end

  private

  def multiple?(number)
    @numbers.any? { |n| (number % n).zero? }
  end
end
